packages/core/database/lib/query/query-builder.js

```diff
@@ -302,9 +302,11 @@ const createQueryBuilder = (uid, db) => {
         }
         case 'count': {
           const dbColumnName =
-            state.count === '*' ? '*' : this.aliasColumn(helpers.toColumnName(meta, state.count));
+            state.count === '*'
+              ? this.aliasColumn(helpers.toColumnName(meta, 'id'))
+              : this.aliasColumn(helpers.toColumnName(meta, state.count));
 
-          qb.count({ count: dbColumnName });
+          qb.countDistinct({ count: dbColumnName });
```

Comment:
Now we need to do a distinct count only when necessary has it has a big impact on performance and if we don't do deep filtering it is not needed.

I think you can use the same if logic as in the select 'shouldUseDistinct' to optimize this

Reply:
@alexandrebodin I eidted the `shouldUseDistinct` a bit and used to fit my case